docs/spec/special-types.rst

```diff
@@ -218,3 +214,98 @@ subtype of ``type[Base]``::
   def new_pro_user(pro_user_class: type[ProUser]):
       user = new_user(pro_user_class)  # OK
       ...
+
+.. _`union-types`:
+
+Union types
```

Comment:
I see that you moved the section on unions from the "concepts" chapter to the "special types" chapter. To me, it feels like it fits better in the "concepts" chapter. After all, it's a foundational concept, not a "special type".

Reply:
Yeah, I agree this was a bad choice. I've moved it back into the concepts doc.